### ISSUE TITLE
Volumetric rendering for non-ortho boxes

### ISSUE THREAD

**Issue by dkoes** [3dmol/3Dmol.js#439]
Current support doesn't work if file format (e.g. ccp4) specifies a transformation matrix.  Shouldn't be too difficult to support since renders take a model to texture transformation matrix already.

**Comment by jaxmatrix** [comment]
I am working on this issues, please tell me where I can find a `dx data file` and `ccp4 data file` with non-orthogonal boxes definition

**Comment by dkoes** [comment]
One of the reasons this issue remains open is I have yet to see an example in the wild.  You can search the PDB for entries that have non-right angles (https://www.rcsb.org/search?request=%7B%22query%22%3A%7B%22type%22%3A%22group%22%2C%22logical_operator%22%3A%22and%22%2C%22nodes%22%3A%5B%7B%22type%22%3A%22group%22%2C%22logical_operator%22%3A%22and%22%2C%22nodes%22%3A%5B%7B%22type%22%3A%22group%22%2C%22logical_operator%22%3A%22and%22%2C%22nodes%22%3A%5B%7B%22type%22%3A%22terminal%22%2C%22service%22%3A%22text%22%2C%22parameters%22%3A%7B%22operator%22%3A%22less%22%2C%22negation%22%3Afalse%2C%22value%22%3A90%2C%22attribute%22%3A%22cell.angle_alpha%22%7D%2C%22node_id%22%3A0%7D%5D%7D%5D%2C%22label%22%3A%22text%22%7D%5D%2C%22label%22%3A%22query-builder%22%7D%2C%22return_type%22%3A%22entry%22%2C%22request_options%22%3A%7B%22pager%22%3A%7B%22start%22%3A0%2C%22rows%22%3A100%7D%2C%22scoring_strategy%22%3A%22combined%22%2C%22sort%22%3A%5B%7B%22sort_by%22%3A%22score%22%2C%22direction%22%3A%22desc%22%7D%5D%7D%2C%22request_info%22%3A%7B%22src%22%3A%22ui%22%2C%22query_id%22%3A%229d8dc3072dabf611c8059c6aefc04ee2%22%7D%7D) and you will find some examples (https://www.rcsb.org/structure/6bna).

However, this reveals a more important issue - the PDB uses DSN6 files (http://www.uoxray.uoregon.edu/tnt/manual/node104.html) which don't yet support.

**Comment by jaxmatrix** [comment]
Is there any open source or free program that can generate these files? I am familiar with Gaussian and Quantum Espresso but these generate electron density in .cube format.

**Comment by dkoes** [comment]
Which files? I'm not familiar with DSN6 but the PDB has many examples.  For ccp4 and dx the gridData submodule of MDAnalysis can write out these formats:
https://www.mdanalysis.org/GridDataFormats/gridData/formats.html

**Comment by jaxmatrix** [comment]
I meant " can a open source program generate ccp4 or dx files".  I tried generating dx files using [octopus](https://octopus-code.org/wiki/Main_Page) by performing simple [dft calculation](https://octopus-code.org/wiki/Tutorial:Visualization) for benzene and methane. `dx` files generated were having orthogonal grid. I guess it has to something with structure of the cell that we are trying to simulate. I will try to perform a similar calculation for some crystal with non-orthogonal lattice to see if I can get some `dx` files with non-orthogonal grid vectors.

**Comment by jaxmatrix** [comment]
I was successful in generating non-ortho boxes when I simulated [hexagonal boron nitride](https://octopus-code.org/wiki/Tutorial:Wires_and_slabs) (h-BN) in octopus using the following input file 



```
CalculationMode = gs
UnitsOutput = ev_angstrom
ExperimentalFeatures = yes

PeriodicDimensions = 2

Spacing = 0.20*angstrom

BNlength = 1.445*angstrom
a = sqrt(3)*BNlength
L = 40
%LatticeParameters
 a | a | L
%

%LatticeVectors
  1.0 | 0.0       | 0.0
 -1/2 | sqrt(3)/2 | 0.0
  0.0 | 0.0       | 1.0
%
 
%ReducedCoordinates
 'B' | 0.0 | 0.0 | 0.0
 'N' | 1/3 | 2/3 | 0.0
% 

PseudopotentialSet=hgh_lda

LCAOStart=lcao_states 

%KPointsGrid
  12   | 12   | 1
%
KPointsUseSymmetries = yes

Output = wfs + density
OutputFormat = dx + cube 
```
I guess volumetric grids are dependent upon the lattice vector of the molecule/system.

I am attaching electron density plot of h-BN in cube and dx format
[h-BN.zip](https://github.com/3dmol/3Dmol.js/files/6226273/h-BN.zip)

**Comment by dkoes** [comment]
Implemented in 1.7.0
